### PR TITLE
Update cli and abstract rest calls into own module

### PR DESCRIPTION
**Pull request by hardbyte**

### CODE REVIEW

**Comment by hardbyte** [comment]
TODO increase test coverage to include error handling: https://codecov.io/gh/n1analytics/clkhash/src/feature-coverage/clkhash/rest_client.py

File: clkhash/cli.py [hunk 1]
```diff
@@ -274,26 +246,35 @@ def results(project, apikey, run, watch, server, output):
...
-        return requests.get(
-            '{}/api/v1/projects/{}/runs/{}/result'.format(server, project, run),
-            headers={"Authorization": apikey}
-        )
+    def log_status(status):
+        log("State: {}".format(status['state']))
+        log("Stage {} ({}/{})".format(
+            status['current_stage']['description'],
+            status['current_stage'],
+            status['stages']))
```
Comment (nbgl) [inline clkhash/cli.py]: `log_status` could be pulled out of `results`

File: clkhash/rest_client.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    if response.status_code != 200:
+        raise ServiceError("Run Status Error", response)
+    return response.json()
+
+
+def run_get_result_text(server, project, run, apikey):
+    return requests.get(
+        '{}/api/v1/projects/{}/runs/{}/result'.format(server, project, run),
+        headers={"Authorization": apikey}
+    ).text
```
[the remark this answers is not in the document]
Comment (hardbyte) [reply clkhash/rest_client.py]: By garbage you mean it will return the error message verbatim from the server?

File: clkhash/cli.py (hunk 1 above)
Comment (hardbyte) [reply clkhash/cli.py]: Good idea

**Pull request merged**